### PR TITLE
Handle unknown data types in data modeling client

### PR DESCRIPTION
**Pull request by erlendvollset**
- Simplify dm data types loading and ensure we handle unknown fields/types
- Bump version and update changelog

## Description
Please describe the change you have made.

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [ ] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite/client/data_classes/data_modeling/data_types.py [hunk 1]
```diff
@@ -48,78 +46,98 @@ def as_tuple(self) -> tuple[str, str]:
...
-            except TypeError:
-                not_supported = set(data).difference(inspect.signature(type_cls).parameters) - {"type"}
-                logger.warning(
-                    f"For '{type_cls.__name__}', the following properties are not yet supported in the SDK (ignored): "
-                    f"{not_supported}. Try updating to the latest SDK version, or create an issue on Github!"
-                )
-                return cast(Self, type_cls(**rename_and_exclude_keys(data, exclude=not_supported)))
-
-        raise ValueError(f"Invalid type {type_}.")
+        obj: Any
```
Comment (doctrino) [inline cognite/client/data_classes/data_modeling/data_types.py:70]: Nice cleanup

File: cognite/client/data_classes/data_modeling/data_types.py [hunk 2]
```diff
@@ -48,78 +46,98 @@ def as_tuple(self) -> tuple[str, str]:
 @dataclass
 class PropertyType(ABC):
     _type: ClassVar[str]
+    is_list: bool = False
```
Comment (doctrino) [inline cognite/client/data_classes/data_modeling/data_types.py:49]: All property types are now listable?

**Review by doctrino — APPROVED** [review]
Great cleanup and nice that all properties are now listable

File: cognite/client/data_classes/data_modeling/data_types.py (hunk 2 above)
Comment (erlendvollset) [reply cognite/client/data_classes/data_modeling/data_types.py:49]: Yes!

File: cognite/client/data_classes/data_modeling/data_types.py [hunk 3]
```diff
@@ -48,78 +46,98 @@ def as_tuple(self) -> tuple[str, str]:
...
+        unit: UnitReference | None = None
+        if (unit_raw := data.get("unit")) and isinstance(unit_raw, dict):
+            unit = UnitReference.load(unit_raw)
+        return unit
+
     @classmethod
     def load(cls, data: dict) -> Self:
-        if "type" not in data:
-            raise ValueError("Property types are required to have a type")
         type_ = data["type"]
```
Comment (doctrino) [inline cognite/client/data_classes/data_modeling/data_types.py:69]: To handle the case when there is not list passed.
```suggestion
        type_ = data["type"]
        is_list = data.get("list", False)
```

File: cognite/client/data_classes/data_modeling/data_types.py (hunk 1 above)
Comment (haakonvt) [reply cognite/client/data_classes/data_modeling/data_types.py:70]: Why change O(1) lookup via _TYPE_LOOKUP to O(N) if-else search? Seems like a step backwards

Comment (erlendvollset) [reply cognite/client/data_classes/data_modeling/data_types.py:70]: I don't agree. This is not on a performance-critical path, so I prefer type safety and simplicity.

Comment (haakonvt) [reply cognite/client/data_classes/data_modeling/data_types.py:70]: Type safety?

Comment (haakonvt) [reply cognite/client/data_classes/data_modeling/data_types.py:70]: Simplicity 😅 
image

**Pull request merged**